src/relax/analysis/layout_transformation.cc

```diff
@@ -0,0 +1,636 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file relax/analysis/layout_transormation.cc
+ * \brief Analyze the PrimFunc and suggest layout transformation on it's blocks and buffers based on
+ * the user provided layout transformations on it's outputs.
+ */
+#include <tvm/arith/iter_affine_map.h>
+#include <tvm/relax/analysis.h>
+#include <tvm/tir/stmt_functor.h>
+
+namespace tvm {
+namespace relax {
+
+using namespace tir;
+
+using VarSet = std::unordered_set<tir::Var, ObjectPtrHash, ObjectPtrEqual>;
+using VarToBlockIndexMap = std::unordered_map<tir::Var, int, ObjectPtrHash, ObjectPtrEqual>;
+using VarToIterTypeMap = std::unordered_map<tir::Var, IterVarType, ObjectPtrHash, ObjectPtrEqual>;
+
+/********** Helper Functions **********/
+
+/*! \brief Converts a list to an Array */
+template <typename T>
+static inline Array<T> list_to_array(std::list<T> l) {
+  Array<T> array;
+  for (auto& v : l) {
+    array.push_back(v);
+  }
+  return array;
+}
+
+/*! \brief Converts an Array to a list */
+template <typename T>
+static inline std::list<T> array_to_list(Array<T> array) {
+  std::list<T> l;
+  for (auto& v : array) {
+    l.push_back(v);
+  }
+  return l;
+}
+
+/*! \brief Checks if `expr` is dependent on any var in `defs`*/
+static inline bool IsDependentOn(const PrimExpr& expr, const VarSet& vars) {
```

Comment:
Is it same as https://github.com/apache/tvm/blob/82a0d4678f66956bb7fa8038e1eedf8c75827685/include/tvm/tir/analysis.h#L126?